### PR TITLE
Add clamav and scanning

### PR DESCRIPTION
**Pull request by whikloj**
Resolves: #60

The message says that an administrator would be notified, but I did not add sending an email here. As the file is not accepted into the bag, do you still want that to happen and if so what would you like it to say.

For testing this I used the EICAR test file, as in [this example](https://rtcamp.com/tutorials/mail/server/testing/antivirus/)

### CODE REVIEW

File: bagitobjecttransfer/recordtransfer/views.py [hunk 1]
```diff
@@ -609,22 +610,30 @@ def _accept_contents(file_upload):
...
+            'error': 'Unable to scan file for malware',
             'verboseError': gettext(
-                'The file "{0}" was identified to contain malware! This issue '
-                'will be sent to the administrator'
-            ),
-            'clamav': {
-                'reason': reason
-                'status': status
-            }
+                'Unable to scan the file "{0}" due to exception {1}'.format(file_upload, exc)
```
Comment (danloveg) [inline bagitobjecttransfer/recordtransfer/views.py:636]: I don't think we should expose the exception contents to the user. If this occurs in other places we should address those

Comment (whikloj) [reply bagitobjecttransfer/recordtransfer/views.py:636]: Fair enough, so I'm logging this error already. Do you want to start having a way to send a message to administrators? Also, do we accept the files if Clamav stops working?

Comment (danloveg) [reply bagitobjecttransfer/recordtransfer/views.py:636]: You should have accession to the `UploadSession` for the transfer. You can call `remove_session_uploads()` on that session to remove all the files that may have been uploaded to the server

**Comment by whikloj** [comment]
Soooo last thing is that currently this is emailing everytime a user hits the Transfer link while ClamAV is inaccessible. 

So _good_ the administrator will not fail to notice the error, but _bad_ we could be filling their inbox. 

Not sure what the best way to handle this is. Some internal flag that tracks the time the first email was sent and only allows an update email after a configurable amount of time.

ie. 
```python

ERROR_EMAIL_PERIOD = 10 * 60 # Ten minutes in seconds
LAST_EMAIL_TIME = None # datetime or None

clamd_socket = clamd.ClamdNetworkSocket(CLAMAV_HOST, CLAMAV_PORT)
try:
    clamd_socket.ping()
except clamd.ClamdError as exc:
    LOGGER.error("Unable to ping ClamAV", exc_info=exc)
    if LAST_EMAIL_TIME = None or LAST_EMAIL_TIME + datetime.timedelta(seconds=ERROR_EMAIL_PERIOD) <= datetime.datetime.now():
        LAST_EMAIL_TIME = datetime.datetime.now()
        send_admin_clamav_error(self.request.user, exc)
```

File: bagitobjecttransfer/recordtransfer/views.py [hunk 2]
```diff
@@ -240,6 +245,17 @@ def get_context_data(self, form, **kwargs):
...
         return context
 
+    def get(self, request, *args, **kwargs):
+        if self.steps.current == 'acceptlegal' and CLAMAV_ENABLED:
+            clamd_socket = clamd.ClamdNetworkSocket(CLAMAV_HOST, CLAMAV_PORT)
+            try:
+                clamd_socket.ping()
+            except clamd.ClamdError as exc:
+                LOGGER.error("Unable to ping ClamAV", exc_info=exc)
+                send_admin_clamav_error(self.request.user, exc)
```
Comment (danloveg) [inline bagitobjecttransfer/recordtransfer/views.py:255]: In response to your comment on spamming admins with emails - how do you feel about simply removing this feature? I assume we will have a system in our apps when deployed to production that sends an email to _system_ administrators when an ERROR message is found in the logs.

This approach simplifies the app, and handles the case where maybe the site admins are not the actual system admins. In our case, we're both, but that might not be the case for others.

**Review by danloveg — COMMENTED** [review]
This all looks great to me. The only issue I see is that you've linked in a system error template, but I don't see it included in the changes you've committed.

I really like the extraction of `_get_admin_recipient_list` into a function.

File: bagitobjecttransfer/recordtransfer/views.py (hunk 2 above)
Comment (whikloj) [reply bagitobjecttransfer/recordtransfer/views.py:255]: We could also add a flag and just display a warning message on the admin pages. I always worry about not getting these notifications fast enough. But I'll remove the emailing for now, we can revisit notifications later.

Comment (danloveg) [reply bagitobjecttransfer/recordtransfer/views.py]: Sure, that sounds good to me

**Pull request merged**